src/tests/quantity.rs

```diff
@@ -452,5 +452,14 @@ mod float {
 
             Test::assert_eq(&3.3.fract(), &m1.fract::<kilogram>().get::<kilogram>());
         }
+
+        #[cfg(feature = "std")]
+        quickcheck! {
+            #[allow(trivial_casts)]
+            fn hypot(l: V, r: V) -> bool {
+                Test::eq(&l.hypot(r),
+                    &f::Length::new::<meter>(l).hypot(f::Length::new::<meter>(r)).get::<meter>())
+            }
+        }
```

Comment:
Can you change these tests to use the normal (`f::`) base units and the kilometer/kilogram/kelvin (`k::`) base units. Something like the `ge` test  higher up in the file.

https://github.com/iliekturtles/uom/blob/afd4e515cbfb2c430f017af099f2cac9a05dd723/src/tests/quantity.rs#L172-L181

Reply:
I think those tests do not really apply to the current signature
```
fn hypot(self, other: Self) -> Self;
```
explicitly prevents mixing base units. Do you want me to make this generic and switch to something like
```
fn hypot(&self, other: &Quantity<D, U1, V>) -> Self;
```
by inserting `change_base` into the implementation? Should this then also depend on the `autoconvert` feature?

Reply:
Good point, I missed that the signature currently requires the same base units. I think it should be switched to accept differing base units based on the `autoconvert` flag. Thoughts?

Reply:
Add both variants of the signature depending on the `autoconvert` feature and extend the test to cover mixing base units.